developer/js/tests/test-error-logger.ts

```diff
@@ -0,0 +1,32 @@
+import { MAX_MESSAGES, KeymanCompilerError, log } from "../dist/errors"
+import { LogHoarder } from "./helpers"
+import { assert } from "chai"
+
+describe('error logger', function () {
+  beforeEach(function () {
+    this.logHoarder = (new LogHoarder).install()
+  })
+
+  afterEach(function () {
+    this.logHoarder.uninstall();
+    delete this.logHoarder;
+  })
+
+  it('should stop logging messages **after** a maximum', function () {
+    for (let i = 0; i < MAX_MESSAGES; i++) {
+      log(KeymanCompilerError.CWARN_DuplicateWordInSameFile, "fake error");
+    }
+
+    // We've logged *just enough messages. This error should not be found:
+    assert.isFalse(this.logHoarder.hasSeenCode(
+      KeymanCompilerError.CWARN_TooManyErrorsOrWarnings
+    ));
+
+    // Log just one too many:
+    log(KeymanCompilerError.CWARN_DuplicateWordInSameFile, "fake error");
+
+    assert.isTrue(this.logHoarder.hasSeenCode(
+      KeymanCompilerError.CWARN_TooManyErrorsOrWarnings
+    ));
```

Comment:
I guess to really test the maximum we should attempt to add another warning after this and validate that it doesn't get added? 🦕